core/tutorials/Unloading/Plugin/PluginClass.cs

```diff
@@ -7,19 +7,16 @@ namespace Plugin
 {
     public class PluginClass : Interface
     {
-        static GCHandle pluginHandle;
-
         public static Interface GetInterface()
         {
             PluginClass plugin = new PluginClass();
 
-            // We create the GC handle just to demonstrate how we can do necessary cleanup before
-            // unloading. Creating a normal GC handle for the plugin instance would prevent unloading
-            // from succeeding.
-            pluginHandle = GCHandle.Alloc(plugin);
-
-            // So we register handler for the Unloading event of the context that we are running in and 
-            // free the GC handle in there.
+            // We register handler for the Unloading event of the context that we are running in 
+            // so that we can perform cleanup of stuff that would otherwise prevent unloading
+            // (Like freeing GCHandles for objects of types loaded into the unloadable AssemblyLoadContext,
+            // terminating threads running code in assemblies loaded into the unloadable AssemblyLoadContext,
```

Comment:
Terminating threads is also a good one. You could start of a work thread here and then in the clean callback set an event and call `Join()` if you were looking for a more explicit clean up scenario. Other than that LGTM.